src/HttpBodyStream.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Legionth\React\Http;
+
+use Psr\Http\Message\StreamInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+use Evenement\EventEmitter;
+use React\Stream\Util;
+
+/**
+ * Uses a StreamInterface from PSR-7 and a ReadableStreamInterface from ReactPHP
+ * to use PSR-7 objects and use ReactPHP streams
+ */
+class HttpBodyStream extends EventEmitter implements StreamInterface, ReadableStreamInterface
+{
+    private $input;
+    private $closed = false;
+    private $encoder;
+
+    public function __construct(ReadableStreamInterface $input, ReadableStreamInterface $encoder = null)
+    {
+        $this->input = $input;
+
+        if ($encoder === null) {
+            $encoder = new ChunkedEncoderStream($this->input);
+        }
+        $this->encoder = $encoder;
+
+        $this->encoder->on('data', array($this, 'handleData'));
+        $this->encoder->on('end', array($this, 'handleEnd'));
+        $this->encoder->on('error', array($this, 'handleError'));
+        $this->encoder->on('close', array($this, 'close'));
+    }
+
+    public function getEncoder()
+    {
+        return $this->encoder;
+    }
+
+    /**
+     * Emits the data
+     * @param string $data - data to be emitted
+     */
+    public function handleData($data)
+    {
+        $this->emit('data', array($data));
+    }
+
+    /**
+     * Handles occuring exceptions on the stream
+     * @param \Exception $e
+     */
+    public function handleError(\Exception $e)
+    {
+        $this->emit('error', array($e));
+        $this->close();
+    }
+
+    /**
+     * Handles the end of the stream
+     */
+    public function handleEnd()
+    {
+        if (!$this->closed) {
+            $this->emit('end');
+            $this->close();
+        }
+    }
+
+    public function __toString()
+    {
+        return '';
+    }
+
+    public function detach()
+    {}
```

Comment:
I could always throw a `BadMethodCallException`?

These are the methods from `StreamInterface` so they aren't really needed anywhere. I could write that, but I don't know if that would help.